pointcloud/basic_octree_utils_test.go

```diff
@@ -220,6 +222,12 @@ func validateBasicOctree(t *testing.T, bOct *BasicOctree, center r3.Vector, side
 		test.That(t, bOct.checkPointPlacement(bOct.node.point.P), test.ShouldBeTrue)
 		test.That(t, bOct.size, test.ShouldEqual, 1)
 		size = bOct.size
+
+		p := bOct.node.point.P
+		mpc := bOct.node.maxProb
+		leafMPCMap[p] = mpc
+		_, err := validateProbability(t, bOct, leafMPCMap)
```

Comment:
is this the right place to be conducting this check?
it did not work for me to place it at line 237, right above `return size`

Reply:
I'd prefer to have it be linked into the validate function similar to how you did it for `Set` and have it be a return value from the `validateOctree` function. 

See on line 211 how I incremented the size based on the size of the children. I think a similar check where you take the max of all the children probs and compare that to the max in the internal node would be ideal.